Tests/StarknetTests/Providers/ProviderTests.swift

```diff
@@ -31,6 +31,15 @@ final class ProviderTests: XCTestCase {
         StarknetProvider(starknetChainId: .testnet, url: url)!
     }
 
+    func testRequestWithCustomURLSession() {
+        let starknetChainId = StarknetChainId.testnet
+        let url = Self.devnetClient.rpcUrl
+        let customURLSession = URLSession(configuration: .ephemeral)
+        let starknetProvider = StarknetProvider(starknetChainId: starknetChainId, url: url, urlSession: customURLSession)
+
+        XCTAssertNotNil(starknetProvider)
+    }
```

Comment:
Added example test, but there's no need for it. This is just another constructor, so not much to test here.